tools/cldr-code/src/main/java/org/unicode/cldr/util/NameGetter.java

```diff
@@ -0,0 +1,547 @@
+package org.unicode.cldr.util;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.ibm.icu.text.MessageFormat;
+import com.ibm.icu.text.Transform;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NameGetter {
+
+    private final CLDRFile cldrFile;
+
+    /**
+     * Construct a new NameGetter.
+     *
+     * @param cldrFile must not be null
+     */
+    public NameGetter(CLDRFile cldrFile) {
+        if (cldrFile == null) {
+            throw new RuntimeException("NameGetter must have non-null CLDRFile");
+        }
+        this.cldrFile = cldrFile;
+    }
+
+    private static final String GETNAME_LOCALE_SEPARATOR =
```

Comment:
Might put PATH in the name, also for similar ones.

Reply:
Fixed in latest commit